src/backend/distributed/utils/reference_table_utils.c

```diff
@@ -429,11 +421,10 @@ DeleteAllReferenceTablePlacementsFromNodeGroup(int32 groupId)
 		BlockWritesToShardList(referenceShardIntervalList);
 	}
 
-	foreach(referenceTableCell, referenceTableList)
+	StringInfo deletePlacementCommand = makeStringInfo();
```

Comment:
Seems reasonable to do this indeed, but these kind of interspersed other refactors (same for the `const` ones) make this quite a bit harder to review.